src/arviz_plots/plots/rootogramplot.py

```diff
@@ -0,0 +1,449 @@
+            pp_distribution,
+            backend=backend,
+            **pc_kwargs,
+        )
+    # set reduce_dims by elimination
+    reduce_dims = [
+        dim for dim in pp_distribution.dims if dim not in set(facet_dims).union(set(sample_dims))
+    ]
+
+    if aes_map is None:
+        aes_map = {}
+    else:
+        aes_map = aes_map.copy()
+    aes_map.setdefault("predictive", plot_collection.aes_set)
+    if labeller is None:
+        labeller = BaseLabeller()
+
+    # setting plot_kwargs_dist defaults (for passing to internal plot_dist calls)
+    plot_kwargs_dist = {
+        key: False
+        for key in ("credible_interval", "point_estimate", "point_estimate_text", "title")
+    }
+    # if "remove_axis" in plot_kwargs:
+    plot_kwargs_dist["remove_axis"] = False  # plot_kwargs["remove_axis"]
+
+    # obs distribution calculated outside `if observed` since plotting predictive bars requires it
+    observed_data_group = "observed_data"
+    obs_distribution = process_group_variables_coords(
+        dt,
+        group=observed_data_group,
+        var_names=var_names,
+        filter_vars=filter_vars,
+        coords=coords,
+    )
+
+    # ---------(observed data)-----------
+    # observed data calculations are made outside of and before 'if observed' since predictive also
+    # depends on this computed data (number of bins and top of predictive bars for rootograms)
+
+    # this portion is situated in an out of convention spot becuse obs_hist_dims is required
+    obs_hist_dims, obs_hist_aes, obs_hist_ignore = filter_aes(
+        plot_collection, aes_map, "observed", reduce_dims
+    )
+
+    # use get_bins func from arviz-stats on observed data and then use those bins for
+    # computing histograms for predictive data as well
+    bins = obs_distribution.azstats.get_bins(dims=list(obs_hist_dims) + list(sample_dims))
+
+    obs_stats_kwargs = copy(stats_kwargs.get("observed", {}))
+    obs_stats_kwargs.setdefault("bins", bins)
+
+    obs_hist = obs_distribution.azstats.histogram(
+        dims=list(obs_hist_dims) + list(sample_dims), **obs_stats_kwargs
+    )
+
+    obs_hist.loc[{"plot_axis": "histogram"}] = (obs_hist.sel(plot_axis="histogram")) ** 0.5
+
+    # new_obs_hist with histogram->y and left_edge/right_edge midpoint->x
+    new_obs_hist = xr.Dataset()
+
+    for var_name in list(obs_hist.keys()):
+        left_edges = obs_hist[var_name].sel(plot_axis="left_edges").values
+        right_edges = obs_hist[var_name].sel(plot_axis="right_edges").values
+
+        left_edges = np.array(left_edges)
+        right_edges = np.array(right_edges)
+
+        x = (left_edges + right_edges) / 2
+        y = obs_hist[var_name].sel(plot_axis="histogram").values
+
+        stacked_data = np.stack((x, y), axis=-1)
+        new_var = xr.DataArray(
+            stacked_data, dims=["hist_dim", "plot_axis"], coords={"plot_axis": ["x", "y"]}
+        )
+
+        new_obs_hist[var_name] = new_var
```

Comment:
```suggestion
new_obs_hist = xr.concat(
    (ds.sel(plot_axis=["right_edges", "left_edges"]).sum("plot_axis") / 2, obs_hist.sel(plot_axis="histogram", drop=True)),
    dim="plot_axis",
).assign_coords(plot_axis=["x", "y"])
```

Reply:
Also note that the code in the PR would only have worked when all variables get the same number of bins (which doesn't always happen) because the dimension name is hardcoded to `hist_dim` in all variables of the dataset. As I mentioned in slack, the behaviour has changed in the `get_bins` PR to support this effect. Now each variable gets independent dimensions `hist_dim_mu`, `hist_dim_tau`...